metrics/metrics/src/main/java/io/helidon/metrics/HelidonSimpleTimer.java

```diff
@@ -0,0 +1,257 @@
+import org.eclipse.microprofile.metrics.MetricID;
+import org.eclipse.microprofile.metrics.MetricType;
+import org.eclipse.microprofile.metrics.MetricUnits;
+import org.eclipse.microprofile.metrics.SimpleTimer;
+
+/**
+ * Implementation of {@link SimpleTimer}.
+ */
+final class HelidonSimpleTimer extends MetricImpl implements SimpleTimer {
+    private final SimpleTimer delegate;
+
+    private HelidonSimpleTimer(String type, Metadata metadata, SimpleTimer delegate) {
+        super(type, metadata);
+        this.delegate = delegate;
+    }
+
+    static HelidonSimpleTimer create(String repoType, Metadata metadata) {
+        return create(repoType, metadata, Clock.system());
+    }
+
+    static HelidonSimpleTimer create(String repoType, Metadata metadata, Clock clock) {
+        return create(repoType, metadata, new SimpleTimerImpl(repoType, metadata.getName(), clock));
+    }
+
+    static HelidonSimpleTimer create(String repoType, Metadata metadata, SimpleTimer metric) {
+        return new HelidonSimpleTimer(repoType, metadata, metric);
+    }
+
+    @Override
+    public void update(Duration duration) {
+        delegate.update(duration);
+    }
+
+    @Override
+    public <T> T time(Callable<T> event) throws Exception {
+        return delegate.time(event);
+    }
+
+    @Override
+    public void time(Runnable event) {
+        delegate.time(event);
+    }
+
+    @Override
+    public Context time() {
+        return delegate.time();
+    }
+
+    @Override
+    public long getCount() {
+        return delegate.getCount();
+    }
+
+    @Override
+    public Duration getElapsedTime() {
+        return delegate.getElapsedTime();
+    }
+
+    @Override
+    public void prometheusData(StringBuilder sb, MetricID metricID, boolean withHelpType) {
+        String promName;
+        String name = metricID.getName();
+        String tags = prometheusTags(metricID.getTags());
+        promName = prometheusName(name) + "_total";
+        if (withHelpType) {
+            prometheusType(sb, promName, "counter");
+            prometheusHelp(sb, promName);
+        }
+        sb.append(promName)
+                .append(tags)
+                .append(" ")
+                .append(getCount())
+                .append("\n");
+
+        promName = prometheusNameWithUnits(name + "_elapsedTime", Optional.of(MetricUnits.SECONDS));
+        if (withHelpType) {
+            prometheusType(sb, promName, "gauge");
+            // By spec, no help for the elapsedTime part of SimpleTimer.
+        }
+        sb.append(promName)
+                .append(tags)
+                .append(" ")
+                .append(elapsedTimeInSeconds())
+                .append("\n");
+    }
+
+    @Override
+    public String prometheusValue() {
+        throw new UnsupportedOperationException("Not supported.");
+    }
+
+    @Override
+    public void jsonData(JsonObjectBuilder builder, MetricID metricID) {
+        JsonObjectBuilder myBuilder = JSON.createObjectBuilder()
```

Comment:
Dim recollection here that this is quite expensive to invoke with every call; I seem to recall some other class that can be cached statically to give you a new builder.

Reply:
All of our metric implementations use the same pattern. The `JSON` field (somewhat misnamed) is declared this way:
```
static final JsonBuilderFactory JSON = 
    Json.createBuilderFactory(Collections.emptyMap());
```
IIRC the real performance drain was in the `createBuilderFactory` call but it probably would be good to review _all_ uses of this pattern to see if we can speed things up.

Reply:
This is the way to do it. We need to cache the builder factory. `JsonObjectBuilder` itself is mutable and cannot be used from more than one thread (as it builds a single JSON Object).